tests/integration/assessment/test_workflows.py

```diff
@@ -4,20 +4,52 @@
 from databricks.sdk.retries import retried
 from databricks.sdk.service.iam import PermissionLevel
 
+from databricks.labs.ucx.hive_metastore import TablesCrawler
+
 
 @retried(on=[NotFound, InvalidParameterValue], timeout=timedelta(minutes=8))
-def test_running_real_assessment_job(ws, installation_ctx, make_cluster_policy, make_cluster_policy_permissions):
+def test_running_real_assessment_job(
+    ws,
+    installation_ctx,
+    make_cluster_policy,
+    make_cluster_policy_permissions,
+    make_schema,
+    make_table,
+    sql_backend,
+    inventory_schema,
+):
     ws_group_a, _ = installation_ctx.make_ucx_group()
     cluster_policy = make_cluster_policy()
     make_cluster_policy_permissions(
         object_id=cluster_policy.policy_id,
         permission_level=PermissionLevel.CAN_USE,
         group_name=ws_group_a.display_name,
     )
+
+    # create tables for crawl_tables assertion
+    source_schema = make_schema(catalog_name="hive_metastore")
+    managed_table = make_table(schema_name=source_schema.name)
+    external_table = make_table(schema_name=source_schema.name, external=True)
+    tmp_table = make_table(schema_name=source_schema.name, ctas="SELECT 2+2 AS four")
+    view = make_table(schema_name=source_schema.name, ctas="SELECT 2+2 AS four", view=True)
+    non_delta = make_table(schema_name=source_schema.name, non_delta=True)
+
     installation_ctx.__dict__['include_object_permissions'] = [f"cluster-policies:{cluster_policy.policy_id}"]
     installation_ctx.workspace_installation.run()
 
     installation_ctx.deployed_workflows.run_workflow("assessment")
 
     after = installation_ctx.generic_permissions_support.load_as_dict("cluster-policies", cluster_policy.policy_id)
     assert after[ws_group_a.display_name] == PermissionLevel.CAN_USE
+
+    tables = []
+    for _ in TablesCrawler(sql_backend, inventory_schema, [source_schema.name]).snapshot():
+        tables.append(_.name)
+
+    assert len(tables) == 5
+
+    assert managed_table.name in tables
+    assert external_table.name in tables
+    assert tmp_table.name in tables
+    assert view.name in tables
+    assert non_delta.name in tables
```

Comment:
I agree that we need an integration test for the fast crawler.

That said, the purpose of this test is to verify that the `assessment` workflow runs: it doesn't really verify _what_ the assessment workflow has done because that's a really large set of things and we can't reasonably verify it all here.

As such I think these tests would be best moved into a separate set of integration tests that focus on the fast crawler itself.

For `TablesCrawler` there's already a start in `test_tables.py`: these tests (at the very least) should also be performed against the fast crawler.

Reply:
For integration testing, FasterTableScanCrawler is tested using the larger assessment test rather than just the class. Testing the class individually would require utilizing a remote spark connection with the Databricks workspace (which is a bit more complicated).